rsconnect/bundle.py

```diff
@@ -1196,7 +1197,9 @@ def make_quarto_manifest(
     :return: the manifest and a list of the files involved.
     """
     if environment:
-        extra_files = list(extra_files or []) + [environment.filename]
+        extra_files = list(extra_files or [])
+        if environment.source == 'file':
```

Comment:
If the environment filename `requirements.txt` was generated in memory, adding it to the `extra_files` would result in a `file not found` exception later on when the bundle is created.

Reply:
I guess this is the other code path. I would suggest always relying on the environment object and using `add_buffer` rather than splitting the functionality.